src/lib/Microsoft.Fx.Portability/ApiPortClient.cs

```diff
@@ -234,60 +216,87 @@ public ApiPortClient(IApiPortService apiPortService, IProgressReporter progressR
         /// </summary>
         /// <param name="options">Options to generate report</param>
         /// <returns>A collection of reports</returns>
-        private async Task<MultipleFormatAnalysis> GetAnalysisResultAsync(IApiPortOptions options)
+        private async Task<MultipleFormatAnalysis> GetReportsAsync(IApiPortOptions options)
         {
             var assemblies = options.InputAssemblies?.Keys ?? Array.Empty<IAssemblyFile>();
             var dependencyInfo = _dependencyFinder.FindDependencies(assemblies, _progressReport);
-
-            if (dependencyInfo.UserAssemblies.Any())
+            if (!dependencyInfo.UserAssemblies.Any())
             {
-                AnalyzeRequest request = GenerateRequest(options, dependencyInfo);
+                _progressReport.ReportIssue(LocalizedStrings.NoFilesAvailableToUpload);
 
-                // Create the progress reporter here (instead of within GetResultFromServiceAsync) since the reporter does not work well when run in parallel
-                using (var progressTask = _progressReport.StartTask(LocalizedStrings.AnalyzingCompatibility))
+                return new MultipleFormatAnalysis
                 {
-                    try
-                    {
-                        var results = await _apiPortService.SendAnalysisAsync(request, options.OutputFormats);
+                    Results = Enumerable.Empty<ReportingResultWithFormat>()
+                };
+            }
 
-                        CheckEndpointStatus(results.Headers.Status);
+            var request = GenerateRequest(options, dependencyInfo);
+            var analyzeResponse = await RequestAnalysisAsync(request);
 
-                        return new MultipleFormatAnalysis
-                        {
-                            Info = dependencyInfo,
-                            Request = request,
-                            Results = results.Response
-                        };
-                    }
-                    catch (Exception)
-                    {
-                        progressTask.Abort();
-                        throw;
-                    }
-                }
+            var reportFormats = options.OutputFormats.Select(displayName => AvailableReportFormats[displayName]);
+            if (!reportFormats.Any())
+            {
+                var defaultFormat = await _apiPortService.GetDefaultResultFormatAsync();
```

Comment:
ConfigureAwait(false)

Reply:
Actually.. I'm going to see if we can get that analyzer in and fix any outstanding ones in the next PR.